test/functional/tests/volumes/test_many_lvms_on_many_cores.py

```diff
@@ -0,0 +1,137 @@
+#
+# Copyright(c) 2022 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+#
+import datetime
+import pytest
+
+from api.cas.init_config import InitConfig, opencas_conf_path
+from storage_devices.lvm import Lvm, LvmConfiguration
+
+from api.cas import casadm
+from core.test_run import TestRun
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from test_tools.fio.fio import Fio
+from test_tools.fio.fio_param import ReadWrite, IoEngine, VerifyMethod
+from test_utils.size import Size, Unit
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core1", DiskTypeLowerThan("cache"))
+@pytest.mark.require_disk("core2", DiskTypeLowerThan("cache"))
+def test_many_lvms_on_many_cores():
+    """
+        title: Test for LVM creation on CAS: 1 cache, 4 cores, 4 lvms.
+        description: |
+          Validation of LVM support, LVMs created (4) on CAS device (1 cache, 4 cores).
+        pass_criteria:
+          - CAS devices created successfully.
+          - LVMs created successfully.
+          - FIO with verification ran successfully.
+          - Configuration after reboot match configuration before.
+    """
+    with TestRun.step(f"Create CAS device."):
+        cores = prepare_devices()
+
+    with TestRun.step("Create LVMs on CAS device."):
+        lvm_filters = ["a/.*/", "r|/dev/sd*|", "r|/dev/hd*|", "r|/dev/xvd*|", "r/disk/", "r/block/",
+                       "r|/dev/nvme*|"]
+        pv_num = 4
+        vg_num = 4
+        lv_num = 4
+        cache_num = 1
+        cas_dev_num = 4
+
+        config = LvmConfiguration(lvm_filters, pv_num, vg_num, lv_num, cache_num, cas_dev_num)
+
+        lvms = Lvm.create_specific_lvm_configuration(cores, config)
+
+    with TestRun.step("Run FIO with verification on LVM."):
+        run_fio_on_lvm(lvms)
```

Comment:
This function is called twice with the same parameter. We could initialize `Fio` object once and then just run it twice.